native/NativeSampleApps/RestExplorer/src/com/salesforce/samples/restexplorer/RestExplorerApp.kt

```diff
@@ -0,0 +1,206 @@
+
+        setLogReceiverFactory(object : SalesforceLogReceiverFactory {
+            override fun create(componentName: String): SalesforceLogReceiver = logReceiversByComponentName[componentName] ?: object : SalesforceLogReceiver {
+                override fun receive(
+                    level: Level,
+                    tag: String,
+                    message: String
+                ) {
+                    val resolvedMessage = "$componentName - $message"
+                    when (level) {
+                        OFF -> {}
+                        ERROR -> Log.e("RestExplorerApp", resolvedMessage)
+                        WARN -> Log.w("RestExplorerApp", resolvedMessage)
+                        INFO -> Log.i("RestExplorerApp", resolvedMessage)
+                        DEBUG -> Log.d("RestExplorerApp", resolvedMessage)
+                        VERBOSE -> Log.v("RestExplorerApp", resolvedMessage)
+                    }
+                }
+
+                override fun receive(
+                    level: Level,
+                    tag: String,
+                    message: String,
+                    throwable: Throwable?
+                ) {
+                    val resolvedMessage = "$componentName - $message"
+                    when (level) {
+                        OFF -> {}
+                        ERROR -> Log.e("RestExplorerApp", resolvedMessage, throwable)
+                        WARN -> Log.w("RestExplorerApp", resolvedMessage, throwable)
+                        INFO -> Log.i("RestExplorerApp", resolvedMessage, throwable)
+                        DEBUG -> Log.d("RestExplorerApp", resolvedMessage, throwable)
+                        VERBOSE -> Log.v("RestExplorerApp", resolvedMessage, throwable)
+                    }
+                }
+            }
+        })
+
+        /*
+		 * Uncomment the following lines to enable push notifications in this app. Either implement
+		 * the methods of the anonymous object or replace it with an instance of another class
+		 * implementing 'PushNotificationInterface'. Add your Firebase 'google-services.json' file
+		 * to the 'app' folder of your project.
+		 */
+//        getInstance().pushNotificationReceiver = object : PushNotificationInterface {
+//            override fun onPushMessageReceived(data: Map<String?, String?>?) {
+//            }
+//
+//            override fun supplyFirebaseMessaging(): FirebaseMessaging? = null
+//        }
+    }
+
+    // endregion
+    // region REST Explorer Salesforce SDK Manager
+
+    internal class RestExplorerSDKManager
+
+    /**
+     * A protected constructor.
+     *
+     * @param context The application context
+     * @param mainActivity The activity to be launched after the login flow
+     * @param loginActivity The login activity
+     */
+    private constructor(
+        context: Context,
+        mainActivity: Class<out Activity?>,
+        loginActivity: Class<out Activity?>
+    ) : SalesforceSDKManager(
```

Comment:
Unrelated to this PR entirely, but is this really a good example of how we want apps to use the Mobile SDK in 2024?  @wmathurin @JohnsonEricAtSalesforce

Reply:
That's a great topic!  I would suspect the answer the that is "No".  There are quite a few aspects of the "SDK Manager" that are following older patterns and I reckon this is one.  We should bring this to planning and create a modernization story (or entire epic) for what we want to do with that class.

Specifically in this context and thinking as an app developer, it would be great to not be subclassing a library-provided class or using it in a singleton context.

Let's gather our notes for planning since this will be a topic bigger than a github thread could handle.